projects/packages/connection/src/class-manager.php

```diff
@@ -94,6 +94,7 @@ public static function configure() {
 
 		if ( $manager->is_connected() ) {
 			add_filter( 'xmlrpc_methods', array( $manager, 'public_xmlrpc_methods' ) );
+			add_filter( 'plugins_loaded', array( new Package_Version_Tracker(), 'maybe_update_package_versions' ) );
```

Comment:
```suggestion
			add_filter( 'plugins_loaded', __NAMESPACE__ . '\Package_Version_Tracker::maybe_update_package_versions' );
```

(since we made the method static).

Btw, while in this context the static method makes sense I'd personally prefer the non-static approach as it makes unit testing so much easier (eg the previous mockable approach was beautiful) and in case we want to add more functionality to this class it won't restrict us to make every method we add static.
However, this is just a personal preference - no need to change your logic!

Reply:
I was on the fence about the static methods. Since you have a preference, I changed them back to non-static. I agree that the non-static approach could make future changes significantly easier to test.

I also brought back the mocked `Package_Version_Tracker::update_package_versions_option` method in `test_maybe_update_package_versions`, and added a new test, `test_maybe_update_package_versions_success`, which tests the case when the HTTP request to WPCOM is successful.